tableauserverclient/models/datasource_item.py

```diff
@@ -1,5 +1,6 @@
 import xml.etree.ElementTree as ET
 from .exceptions import UnpopulatedPropertyError
+from .property_decorators import property_not_nullable
```

Comment:
At first glance the code feels ok, but I think there might be a better name than "property_decorator" which both have meanings of their own.

I'll do a more detailed review in a bit